src/operators.rs

```diff
@@ -285,6 +301,46 @@ impl Initializer<FenderReference> for FenderInitializer {
                 }
                 FenderValue::String(collected.into()).into()
             }
+            FenderInitializer::Struct(id) => {
+                match ctx.context.struct_table.type_list().get(*id).cloned() {
+                    Some(struct_type) => {
+                        let mut ret = FenderStruct {
+                            struct_id: struct_type.clone(),
+                            data: HashMap::new(),
+                        };
+                        for (key, type_id, val) in
+                            struct_type.fields.iter().zip(values.into_iter()).map(
+                                |((name, type_id), val)| {
```

Comment:
Is this performing a hashmap lookup with a string whenever structs are initialized/referenced?

Reply:
yeah this is why I didn't know what the point of the converting from the string to a usize was, and I figured I was missing a part of what you were explaining

Reply:
It should all be done at compiletime, so that by runtime all the field names have been converted into usize and you never need to do a string lookup

Reply:
rust compile time? how?
every time you said compile time I was confused cause this would do something if fender was compiled

Reply:
Fender compiletime - find all field names used beforehand and convert them into usize constants before the program starts running

Reply:
It's doing that

Reply:
But this seems to be doing string lookups at runtime

Reply:
oh it is still doing it here, it doesn't do it for references
references are converted to usize when fender is interpreted

Reply:
Can you make it do the same here?

Reply:
yeah